<a href="https://colab.research.google.com/github/nsriniva/DS-Unit-1-Sprint-2-Statistics/blob/master/module3/LS_DS_123_Confidence_Intervals_Assignment_No_Answers.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Confidence Intervals

Soft drinks like Coke and Pepsi are manufactured to have a standard caffeine content. For example, a 12-oz serving of Coke has 34mg of caffeine, and a 12-oz serving of Pepsi has 37.6mg of caffeine. However, fountain soft drinks are typically mixed in individual restaurant dispensers, so it is more difficult to maintain a standard level of caffeine per serving. In this study, researchers randomly sampled Coke, Diet Coke, Pepsi, and Diet Pepsi at a set of franchise restaurants and measured the caffeine content in 12oz of each soft drink2. The data is found in the Soda.xlsx dataset.

Because individuals can be sensitive to caffeine – and because the manufacturers are interested in product consistency – we wish to estimate the mean caffeine content in 12oz of Coke served in franchise restaurants using a 95% confidence interval. 

You can find the Coke data here: https://raw.githubusercontent.com/Chelsea-Myers/Lambda-Intro/master/Coke.csv

The first variable is the sample ID and the second variable is the caffeine content in the 12-oz sample measured in mg.

Source: A.N. Garand and L.N. Bell (1997). "Caffeine Content of Fountain and Private-Label Store Brand Carbonated Beverages," Journal of the American Dietetic Association, Vol. 97, #2, pp. 179-182.




###1) Load the dataset and print the first few rows

In [19]:
#Import statements
import pandas as pd
import numpy as np

coke_url = 'https://raw.githubusercontent.com/Chelsea-Myers/Lambda-Intro/master/Coke.csv'

df = pd.read_csv(coke_url)

In [15]:
#Print the top 5 rows
df.head(5)

Drink  Caffeine
0      1     47.32
1      2     43.78
2      3     48.12
3      4     43.25
4      5     46.42

###2) Calculate the mean, SD, SE caffeiene content and n for the sample.  Summarize your results in a sentence or two.

In [9]:
#Calculations here
c_mean = df.Caffeine.mean()
c_sd   = df.Caffeine.std()
c_n = len(df.Caffeine)

c_se = c_sd/(c_n**0.5)

print(c_mean, c_sd, c_n, c_se)

37.9402 5.243756828216712 50 0.7415792024250598


The mean caffeine content for the sample size of 50 is 37.9402 mg with a standard deviation of 5.2438 and a standard error of 0.7416

###3) Find t* for a 95% confidence interval.  

Use the starter code below and fill in the degrees of freedom.

In [14]:
from scipy.stats import t

#Don't worry too much about where the 0.975 comes from.  It has to do
#with wanting to determine the *middle* 95% of the t-distribution
#We're going to learn
#how to calculate a 95% CI this easy way in just a minute.

#Recall that n = 223 for the body temp problem.
t_star = t.ppf(0.975,df=c_n-1)
print('t_star =', t_star)

t_star = 2.009575234489209


###4) Calculate the margin of error for a 95% confidence interval for the mean caffeine content in a 12-oz Coke.



In [17]:
#Calculations here
err_margin = t_star*c_se
err_margin

1.49031936

The margin of error is $t^**0.7416$ = 2.0096*0.7416 = 1.4903

###5) Calculate a 95% CI for the mean caffeine content in a 12-oz fountain Coke with the CI formula using the summary statistics and t* that you calculated above.

In [18]:
#Calculations here
lower_ci = c_mean - err_margin
upper_ci = c_mean + err_margin

print(lower_ci, upper_ci)

36.44994080039434 39.43045919960566


###6) Calculate a 95% CI for the mean caffeiene content in a 12-oz fountain Coke using the t-interval function in Python.

In [21]:
#Calculations here
l95_ci, u95_ci = t.interval(0.95, df = c_n-1,loc=c_mean, scale=c_se )
print(l95_ci, u95_ci)

36.44994080039434 39.43045919960566


###7) Compare the two confidence intervals you calculated.  Do they match?  Should they?

Both the confidence intervals are identical, as they should be.

###8) Interpret the meaning of the 95% confidence interval for the mean caffeiene content in a 12-oz fountain Coke. in a sentence or two.

We are 95% confident that mean caffeine content in 12oz of Coke served in franchise restaurants is between 36.45 and 39.43 mg.

###9) Using the t-interval Python function, calculate a 90% confidence interval for the mean caffeine content in a 12-oz Coke.  Is this estimate more accurate or more precise (pick one) than the 95% confidence interval?



In [22]:
#Calculations here
l90_ci,u90_ci = t.interval(0.9, df = c_n-1,loc=c_mean, scale=c_se)
print(l90_ci, u90_ci)

36.696904726749196 39.1834952732508


We are 90% confident that mean caffeine content in 12oz of Coke served in franchise restaurants is between 36.70 and 39.18 mg.

This estimate is less(90% as opposed to 95%) accurate but more precise(smaller range).

###10) Using the t-interval Python function, calculate a 99% confidence interval for the mean caffeine content in a 12-oz Coke.  Is this estimate more accurate or more precise (pick one) than the 95% confidence interval?



In [23]:
#Calculations here
l99_ci,u99_ci = t.interval(0.99, df = c_n-1,loc=c_mean, scale=c_se)
print(l99_ci, u99_ci)

35.95280335285685 39.92759664714315


We are 99% confident that mean caffeine content in 12oz of Coke served in franchise restaurants is between 35.95 and 39.93 mg.

This estimate is more(99% as opposed to 95%) accurate but less precise(larger range).


## Stretch goals:

###1) The correspondence between confidence intervals and hypothesis tests.

Read [this](https://https://blog.minitab.com/blog/adventures-in-statistics-2/understanding-hypothesis-tests-confidence-intervals-and-confidence-levels#:~:text=If%20a%20hypothesis%20test%20produces,corresponding%20confidence%20level%20is%2095%25.&text=If%20the%20confidence%20interval%20does,the%20results%20are%20statistically%20significant.) article about the correspondence between confidence intervals and hypothesis tests.  Feel free to read the whole article, but the relevant part can be found under the heading Why P Values and Confidence Intervals Always Agree About Statistical Significance.

Imagine you work for quality control at Coke and are tasked with making sure that the caffeiene content in the fountain beverages served in restaurants is the same as in a 12-oz can of Coke (34mg).  If you believe that the mean caffeiene content in fountain coke is not 34mg, you must re-train the franchise managers to make sure the Coke served has the correct caffeiene level.

Based on the confidence interval you calculated in the assignment, do you believe that the mean caffeiene content is statistically significantly different from 34 mg in a 12-oz serving?


Answer here.

###2) If we increased the sample size from 50 to 100 but the sample mean and SD remained the same, describe **two** ways the margin of error would change.  Would the margin of error become smaller or larger?

Answer to what would change here.

In [ ]:
#Calculations here.

Compare MEs here.

